be/src/exec/es/es_predicate.cpp

```diff
@@ -0,0 +1,415 @@
+        vector<ExtLiteral> query_conditions;
+        query_conditions.emplace_back(literal);
+        vector<ExtColumnDesc> cols; //TODO
+        ExtPredicate* predicate = new ExtFunction(
+                        TExprNodeType::FUNCTION_CALL,
+                        conjunct->fn().name.function_name,
+                        cols,
+                        query_conditions);
+        if (_es_query_status.ok()) {
+            _es_query_status 
+                = BooleanQueryBuilder::check_es_query(*(ExtFunction *)predicate); 
+            if (!_es_query_status.ok()) {
+                delete predicate;
+                return _es_query_status;
+            }
+        }
+        _disjuncts.push_back(predicate);
+
+        return Status::OK;
+    } 
+
+    if (TExprNodeType::FUNCTION_CALL == conjunct->node_type()) {
+        std::string fname = conjunct->fn().name.function_name;
+        if (fname != "like") {
+            return Status("build disjuncts failed: function name is not like");
+        }
+
+        SlotRef* slotRef = nullptr;
+        Expr* expr = nullptr;
+        if (TExprNodeType::SLOT_REF == conjunct->get_child(0)->node_type()) {
+            expr = conjunct->get_child(1);
+            slotRef = (SlotRef*)(conjunct->get_child(0));
+        } else if (TExprNodeType::SLOT_REF == conjunct->get_child(1)->node_type()) {
+            expr = conjunct->get_child(0);
+            slotRef = (SlotRef*)(conjunct->get_child(1));
+        } else {
+            return Status("build disjuncts failed: no SLOT_REF child");
+        }
+
+        const SlotDescriptor* slot_desc = get_slot_desc(slotRef);
+        if (slot_desc == nullptr) {
+            return Status("build disjuncts failed: slot_desc is null");
+        }
+
+        PrimitiveType type = expr->type().type;
+        if (type != TYPE_VARCHAR && type != TYPE_CHAR) {
+            return Status("build disjuncts failed: like value is not a string");
+        }
+
+        ExtLiteral literal(type, _context->get_value(expr, NULL));
+        ExtPredicate* predicate = new ExtLikePredicate(
+                    TExprNodeType::LIKE_PRED,
+                    slot_desc->col_name(),
+                    slot_desc->type(),
+                    literal);
+
+        _disjuncts.push_back(predicate);
+        return Status::OK;
+    }
+      
+    if (TExprNodeType::IN_PRED == conjunct->node_type()) {
+        // the op code maybe FILTER_NEW_IN, it means there is function in list
+        // like col_a in (abs(1))
+        if (TExprOpcode::FILTER_IN != conjunct->op() 
+            && TExprOpcode::FILTER_NOT_IN != conjunct->op()) {
+            return Status("build disjuncts failed: "
+                        "opcode in IN_PRED is neither FILTER_IN nor FILTER_NOT_IN");
+        }
+
+        vector<ExtLiteral> in_pred_values;
+        const InPredicate* pred = dynamic_cast<const InPredicate*>(conjunct);
+        if (Expr::type_without_cast(pred->get_child(0)) != TExprNodeType::SLOT_REF) {
```

Comment:
add a function to get `expr` without cast

Reply:
Done